infra/feast-operator/internal/controller/services/services.go

```diff
@@ -485,6 +485,25 @@ func (feast *FeastServices) getServiceConfigs(feastType FeastServiceType) feastd
 	return feastdevv1alpha1.ServiceConfigs{}
 }
 
+func (feast *FeastServices) getServiceReplicas(feastType FeastServiceType) *int32 {
+	appliedServices := feast.Handler.FeatureStore.Status.Applied.Services
+	switch feastType {
+	case OfflineFeastType:
+		if feast.isOfflinStore() {
+			return appliedServices.OfflineStore.Replicas
+		}
+	case OnlineFeastType:
+		if feast.isOnlinStore() {
+			return appliedServices.OnlineStore.Replicas
+		}
+	case RegistryFeastType:
+		if feast.isLocalRegistry() {
+			return &DefaultReplicas
+		}
+	}
```

Comment:
is this section necessary?
```suggestion
```

Reply:
I guess not, since it gets covered by the default return value at the end. Committed.